src/tools/FreeDraw.ts

```diff
@@ -1,47 +1,66 @@
 import { Tool } from '@/tools/Tool'
 import Konva from 'konva'
+import { CanvasElementPayload, CanvasElement } from '@/types/Canvas'
+import uuid from 'uuid'
+import throttle from 'lodash.throttle'
 
 export default class FreeDraw implements Tool {
-  // eslint-disable-next-line no-useless-constructor
+  private line: Konva.Line;
   constructor (public readonly name: string,
                public size: number,
                public colour: string) {
+    this.line = new Konva.Line({ points: [0, 0] })
   }
 
-  action (e: Konva.KonvaPointerEvent, stage: Konva.Stage | null, layer: Konva.Layer): void {
-    // created canvas we can add to layer as "Konva.Image" element
-    const line = new Konva.Line({
-      stroke: this.colour,
-      strokeWidth: Number(this.size),
-      lineJoin: 'round',
-      globalCompositeOperation: 'source-over',
-      points: [e.evt.x, e.evt.y]
-    })
-    layer.add(line)
-
-    const newPoints = line.points().concat([e.evt.x + 3, e.evt.y + 3])
-    line.points(newPoints)
+  mouseDownAction = (e: Konva.KonvaPointerEvent, canvasElementPayload: CanvasElementPayload, layer: Konva.Layer, _socket: WebSocket): void => {
+    canvasElementPayload.data = []
+    canvasElementPayload.data.push({ x: e.evt.x, y: e.evt.y })
+    this.line = this.createElement(canvasElementPayload)
```

Comment:
what's the point of having a line class member if you reassign a new value everytime? A local variable suffice here

Reply:
Good point. i will correct this

Reply:
Upon reflection we only assign a new Line when the mouse is first pressed down, afterwards we use the existing line class.

Reply:
The one in the constructor is just a basic template to get around having to set line as undefined.